unlock-app/src/components/interface/locks/Settings/forms/UpdateReferralFee.tsx

```diff
@@ -105,19 +133,76 @@ export const UpdateReferralFee = ({
             errors?.referralFeePercentage &&
             'This field accept percentage value between 0 and 100.'
           }
-          disabled={disabledInput || !referralFee}
+          disabled={isDisabledReferrerInput || !isReferralFeeEnabled}
         />
       </div>
+
+      <div className="grid gap-2">
+        <ToggleSwitch
+          title="Referrer address"
+          enabled={isReferralAddressEnabled}
+          disabled={isDisabledReferrerInput}
+          setEnabled={(enabled: boolean) => {
+            setIsReferralAddressEnabled(enabled)
+          }}
+        />
+        <Input
```

Comment:
Please use our AddressInput component so that it does ENS resolution.

Reply:
Updated 👍 [🧼 Resolve ENS name with <AddressInput /> component](https://github.com/unlock-protocol/unlock/pull/12574/commits/1daac8f8281271eb402a3e0dc4ad168fcfc133a2)